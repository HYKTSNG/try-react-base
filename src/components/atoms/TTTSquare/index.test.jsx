import { render } from "@testing-library/react"
import React from "react"

import { TTTSquare } from "./"

describe("snapshot testing", () => {
  test("no value", () => {
    // ## Arrange ##
    // ## Act ##
    const { container } = render(<TTTSquare />)

    // ## Assert ##
    expect(container).toMatchSnapshot()
  })

  test("O", () => {
    // ## Arrange ##
    // ## Act ##
    const { container } = render(<TTTSquare value="O" />)

    // ## Assert ##
    expect(container).toMatchSnapshot()
  })

  test("disabled", () => {
    // ## Arrange ##
    // ## Act ##
    const { container } = render(<TTTSquare disabled value="X" />)

    // ## Assert ##
    expect(container).toMatchSnapshot()
  })
})
