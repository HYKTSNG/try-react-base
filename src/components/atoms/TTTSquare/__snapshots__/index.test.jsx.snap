// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing O 1`] = `
<div>
  <button
    class="css-cvtux9-TTTSquare"
  >
    O
  </button>
</div>
`;

exports[`snapshot testing disabled 1`] = `
<div>
  <button
    class="css-cvtux9-TTTSquare"
    disabled=""
  >
    X
  </button>
</div>
`;

exports[`snapshot testing no value 1`] = `
<div>
  <button
    class="css-cvtux9-TTTSquare"
  />
</div>
`;
